app/models/user.rb

```diff
@@ -163,6 +163,14 @@ def internal?
     auth_source.is_a? AuthSourceInternal
   end
 
+  def ldap?
+    auth_source.is_a? AuthSourceLdap
+  end
+
+  def can_update_username?
+    !ldap?
```

Comment:
:+1:, esp as LDAP is not the only source we can't update. External (Kerberos) based auth is also quite strict about that ;)